scalanet/src/io/iohk/scalanet/peergroup/UDPPeerGroup.scala

```diff
@@ -310,18 +277,17 @@ class UDPPeerGroup[M](val config: Config, cleanupScheduler: Scheduler = Schedule
     ct.map { nettyChannel =>
         val localAddress = nettyChannel.localAddress()
         log.debug(s"Generated local address for new client is $localAddress")
-        val channelId = getChannelId(to.inetSocketAddress, localAddress)
-
-        assert(!activeChannels.containsKey(channelId), s"HOUSTON, WE HAVE A MULTIPLEXING PROBLEM")
-
         val channel = new ChannelImpl(
           nettyChannel,
           localAddress,
           to.inetSocketAddress,
           ConnectableSubject[ChannelEvent[M]](),
           UDPPeerGroupInternals.ClientChannel
         )
-        activeChannels.put(channelId, channel)
+        // By using netty channel id as part of our channel id, we make sure that each client channel is unique
+        // therefore the won't such channels in active channels map already.
+        activeChannels.put(channel.channelId, channel)
+        channel.closePromise.addListener(closeChannelListener)
```

Comment:
should we add the `channel created` message here?

Reply:
`ChannelCreated` event is only used for server channels, in case of client channel user of library gets reference to this new channel directly from client method - `Task[Channel[InetMultiAddress, M]]`